Eradication/VectorPopulation.cpp

```diff
@@ -2846,49 +2795,39 @@ namespace Kernel
                                                    VectorCohortVector_t* pMigratingQueue,
                                                    VectorCohortCollectionAbstract& rQueue )
     {
+
         Gender::Enum             human_gender_equivalent = m_pMigrationInfoVector->ConvertVectorGender( vector_gender );
-        const std::vector<suids::suid>& reacheable_nodes = m_pMigrationInfoVector->GetReachableNodes( human_gender_equivalent );
-        std::vector<float>            fraction_traveling = m_pMigrationInfoVector->GetFractionTraveling( vector_gender, 0 );
-        // we could use FractionTraveling to also set the GetTotalFemale and Male, but for that we'd have to keep track of male totals with a new parameter
+        const std::vector<suids::suid>  reacheable_nodes = m_pMigrationInfoVector->GetReachableNodes( human_gender_equivalent );
+        std::vector<float>            fraction_traveling = m_pMigrationInfoVector->GetFractionTraveling( vector_gender, 0 ); // index = 0 is the rate for non-allele travel
 
-        if( m_pMigrationInfoVector->GetMigrationAlleleCombinationsSize() == 1 )
+        if( !m_pMigrationInfoVector->TravelByAlleles() )
         {
-            // if this is generic vector migration and total rate is 0, be done
-            float total_traveling = 0;
-            for( float fraction : fraction_traveling ) {
-                total_traveling += fraction;
-            }
-            if( total_traveling == 0 ) // should we also kick out at tiny numbers ?
+            if( m_pMigrationInfoVector->GetTotalRate(human_gender_equivalent) == 0)
```

Comment:
We want this to be more abstract.  This should know about traveling by alleles.  Maybe a CanTravel( Gender::Enum gender ) version that does these two things.

discovered below that we can probably make the argument VectorGender::Enum

Reply:
I combined CanTravel and MightTravel functions into a MightTravel which checks 1. If we have a null object so there's no migration 2. if we have non-allele migration AND totalRate is 0.